api/src/main/java/com/codeforcommunity/rest/RestFunctions.java

```diff
@@ -51,16 +53,34 @@ static int getRequestParameterAsInt(HttpServerRequest req, String name) {
     }
   }
 
-  static String getRequestParameterAsString(HttpServerRequest req, String name) {
+  public static String getRequestParameterAsString(HttpServerRequest req, String name) {
     String paramValue = req.getParam(name);
     if (paramValue != null && !paramValue.isEmpty()) {
       return paramValue;
     }
     throw new MissingParameterException(name);
   }
 
-  static boolean getRequestParameterAsBoolean(HttpServerRequest req, String name) {
+  public static boolean getRequestParameterAsBoolean(HttpServerRequest req, String name) {
     String paramValue = req.getParam(name);
     return Boolean.parseBoolean(paramValue);
   }
+
+  public static long getPathParamAsLong(RoutingContext ctx, String paramName) {
+    try {
+      String paramValue = ctx.pathParam(paramName);
+      return Long.parseLong(paramValue);
+    } catch (NumberFormatException ex) {
+      throw new MalformedParameterException(paramName);
+    }
+  }
+
+  public static Country getCountryFromString(String countryName) {
```

Comment:
Should this be defined in the enum file?

Reply:
The issue is that our exceptions are defined in `api`, meaning that if we want to throw a custom exception that can't happen in `common`, but in this situation we always want to throw a custom exception. Any thoughts?